Sources/Decoder.swift

```diff
@@ -157,6 +157,9 @@ public struct Decoder {
                 
                 for subJSON in jsonArray {
                     guard let model = T(json: subJSON) else {
+                        if T.self is ArrayPartiallyDecodable.Type {
```

Comment:
ah - sorry for being dense. 

i don't think it makes sense for an end user of this library that only simple types and enum types would be partially Decodable - one would expect any type to be so. what do you think?